src/services/actions/userActions.ts

```diff
@@ -32,3 +35,108 @@ export async function makeRegularUser(user: { userId: string }) {
     revalidatePath('/', 'layout')
   }
 }
+
+export async function updateUserProfile(
+  data: ProfileDataValues,
+) {
+   try {
```

Comment:
2 things are missing.

1. u should always validate the input data by using zod.parse. Checkout other actions.
2. u must validate that the current user is the owner of that profile.